field/src/extension/complex.rs

```diff
@@ -56,6 +56,14 @@ impl<AF: AbstractField> Complex<AF> {
     pub fn to_array(&self) -> [AF; 2] {
         self.value.clone()
     }
+    // Sometimes we want to rotate over an extension that's not necessarily ComplexExtendable,
+    // but still on the circle.
+    pub fn rotate<Ext: AbstractExtensionField<AF>>(&self, rhs: Complex<Ext>) -> Complex<Ext> {
+        Complex::<Ext>::new(
+            rhs.real() * self.real() - rhs.imag() * self.imag(),
```

Comment:
nit: even though this is not a hardcoded convention, `self` is typically written first.

Reply:
This is necessary because `rhs` is an extension of `self`, so we are using its `Mul<Base>` overload (it will not compile the other way around)

Reply:
Oh right, makes sense !